nntrainer/layers/layer_node.cpp

```diff
@@ -95,69 +121,53 @@ int LayerNode::setProperty(std::vector<std::string> properties) {
       /// @note this calls derived setProperty if available
       setProperty(static_cast<nntrainer::LayerV1::PropertyType>(type), value);
     } catch (...) {
-      remainder.push_back(properties[i]);
+      remainder.push_back(left_properties[i]);
     }
   }
 
-  status = getLayer()->setProperty(remainder);
+  /// note that setting distribute is only allowed for one time.
+  /// until we have layerNode::finalize and must not except timedist layer
+  if (getDistribute()) {
+    if (layerv1 == nullptr) {
+      /// logic for layer v2
+    } else {
+      auto &ac = nntrainer::AppContext::Global();
+      std::shared_ptr<nntrainer::LayerV1> dlayer =
+        ac.createObject<nntrainer::LayerV1>(TimeDistLayer::type);
+      std::static_pointer_cast<TimeDistLayer>(dlayer)->setDistLayer(layerv1);
+      layerv1 = dlayer;
+    }
+  }
+
+  if (layerv1 == nullptr) {
+    layer->setProperty(remainder);
+  } else {
+    auto &l = getLayer();
+    return l->setProperty(remainder);
```

Comment:
why is the behavior different than LayerV2?

Reply:
V2 is void operation, V1 returns int.

Reply:
So, if line 143 fails, it will throw

Reply:
aah got it.